backend/tests/test_models.py

```diff
@@ -18,20 +23,132 @@ def test_delete_lesson_with_exercises(self, db_session, lesson_with_exercise):
+    def test_progress_cascade_deleted_for_lesson(self, db_session):
         """
         Progress records should be deleted if their lesson object was deleted.
         """
-        pass
 
-    def test_progress_cascade_deleted_for_user(self):
+        user = User(
+            id=1,
+            email=faker.email(),
+            background=faker.color(),
+            emoji=faker.color(),
+            password=faker.password(),
+            streak=1,
+        )
+
+        db_session.add(user),
+        db_session.commit(),
+        db_session.refresh(user)
+
+        lesson = Lesson(
+            name=faker.name(),
+            content=faker.text(),
+            preview=faker.url(),
+            order=0,
+        )
+        db_session.add(lesson)
+        db_session.commit()
+        db_session.refresh(lesson)
+
+        progress = Progress(
+            user_id=user.id,
+            lesson_id=lesson.id,
+        )
+
+        db_session.add(progress)
+        db_session.commit()
+
+        assert progress
+
+        db_session.delete(lesson)
+        db_session.commit()
+
+        try:
+            assert not progress.lesson_id
+        except ObjectDeletedError:
+            db_session.delete(user)
+            db_session.commit()
+            return 0
+
+    def test_progress_cascade_deleted_for_user(self, db_session):
         """
         Progress records should be deleted if their user object was deleted.
         """
-        pass
+        user = User(
+            id=1,
+            email=faker.email(),
+            background=faker.color(),
+            emoji=faker.color(),
+            password=faker.password(),
+            streak=1,
+        )
+
+        db_session.add(user),
+        db_session.commit(),
+        db_session.refresh(user)
+
+        lesson = Lesson(
+            name=faker.name(),
+            content=faker.text(),
+            preview=faker.url(),
+            order=0,
+        )
+        db_session.add(lesson)
+        db_session.commit()
+        db_session.refresh(lesson)
+
+        progress = Progress(
+            user_id=user.id,
+            lesson_id=lesson.id,
+        )
+
+        db_session.add(progress)
+        db_session.commit()
+
+        assert progress
+
+        db_session.delete(user)
+        db_session.commit()
 
-    def test_lesson_order_unique(self):
+        try:
+            assert not progress.lesson_id
+        except ObjectDeletedError:
+            db_session.delete(lesson)
+            db_session.commit()
+            return 0
+
+    def test_lesson_order_unique(self, db_session):
         """
         Lessons should have unique order number.
         """
-        pass
+
+        lesson1 = Lesson(
+            name=faker.name(),
+            content=faker.text(),
+            preview=faker.url(),
+            order=1,
+        )
+        db_session.add(lesson1)
+        db_session.commit()
+        db_session.refresh(lesson1)
+
+        try:
```

Comment:
pytest has a better solution for testing expected exceptions instead of try except: https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions